### ISSUE TITLE
Feature Request: List mulled environment in Job Dependencies section

### ISSUE THREAD

**Issue by hexylena** [galaxyproject/galaxy#5033]
Trying to debug some failed jobs and I don't know which mulled environment corresponds to a set of dependencies. Some tools (e.g. bowtie) show this in the tool_version information. 

![utvalg_318](https://user-images.githubusercontent.com/458683/32942031-ac05a128-cb7f-11e7-9db8-e2717ef25976.png)

Others do not show it at all. I'd love something like:

![utvalg_319](https://user-images.githubusercontent.com/458683/32942025-a6ec58e4-cb7f-11e7-851d-c074a8793c5e.png)

**Comment by mvdbeek** [comment]
We could predict the environment that would be used, or we could record this with the JobParameters in the database (other ideas?). I'm not sure what would be best. You can also see which environment is being used in the admin panel, under Manage tool dependencies, but that's a lot of clicks from a failed job ...

**Comment by bgruening** [comment]
I fear that the reverse lookup will not be possible. But we can provide the mulled encoding for an arbitrary set of requirements. Code for this should be in galaxy-lib.

**Comment by mvdbeek** [comment]
Yes, you can't go from mulled env to individual requirements without a reversible hash. But the job records the tool, the tool record the requirements, and from there you can get the mulled environment. Which is what we already do in the Manage tool dependencies page.

**Comment by nsoranzo** [comment]
As you point out above, that may be different with respect to what was used when the job was actually run (e.g. if the dependencies were installed later).

**Comment by mvdbeek** [comment]
Right, but the initial question `which mulled environment corresponds to a set of dependencies`, we can answer unambiguously (as long as the combination of requirements is still present in one of the tools).

It would also be great to either record the mulled environment, or even to record the contents of the mulled environment, which we could collect together with the job metadata.
